targets/frontend/src/pages/storage/[container]/index.js

```diff
@@ -0,0 +1,327 @@
+  IoMdCheckmark,
+  IoMdClipboard,
+  IoMdCloseCircleOutline,
+  IoMdDownload,
+} from "react-icons/io";
+import { Button, IconButton } from "src/components/button";
+import { Layout } from "src/components/layout/auth.layout";
+import { DropZone } from "src/components/storage/DropZone";
+import { withCustomUrqlClient } from "src/hoc/CustomUrqlClient";
+import { withUserProvider } from "src/hoc/UserProvider";
+import { useDebouncedState } from "src/hooks/";
+import { deleteFile, listFiles, uploadFiles } from "src/lib/azure";
+import { timeSince } from "src/lib/duration";
+import useSWR, { mutate } from "swr";
+import {
+  Card,
+  Field,
+  Flex,
+  jsx,
+  Label,
+  Message,
+  Select,
+  Spinner,
+} from "theme-ui";
+
+const onDeleteClick = function (container, file) {
+  const confirmed = confirm(
+    `Êtes-vous sûr(e) de vouloir définitivement supprimer ${file.name} ?`
+  );
+  if (confirmed) {
+    deleteFile(container, file.name)
+      .then(() => {
+        mutate("files");
+      })
+      .catch(() => {
+        alert("Impossible de supprimer le fichier :/");
+      });
+  }
+};
+
+function FilesPage() {
+  const router = useRouter();
+  const container = router.query.container;
+  const { data = [], error } = useSWR("files", listFiles(container));
+
+  const [hasClipboardApi, setHasClipboardApi] = useState(false);
+  const [currentClip, setCurrentClip] = useState("");
+  const [isSearching, setSearching] = useState(false);
+  const [search, setSearch] = useState("");
+  const [sort, setSort] = useState("mostRecent");
+  const [filter, setFilter] = useState("");
+  const [
+    displayedFiles,
+    setDisplayedFiles,
+    setDebouncedDisplayedFiles,
+  ] = useDebouncedState([], 400);
+
+  // Maybe we should limit displayed files to a number. Like 50 ?
+  useEffect(() => {
+    // prevent loader at first display of page
+    if (!displayedFiles.length && !search)
+      return setDisplayedFiles(
+        data.filter(filterCallback(filter)).sort(getSortCallback(sort))
+      );
+    setSearching(true);
+    setDebouncedDisplayedFiles(
+      data
+        .filter((file) =>
+          file.name.toLowerCase().includes(search.toLowerCase().trim())
+        )
+        .filter(filterCallback(filter))
+        .sort(getSortCallback(sort))
+    );
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    data,
+    sort,
+    filter,
+    search,
+    setDisplayedFiles,
+    setDebouncedDisplayedFiles,
+  ]);
+
+  useEffect(() => {
+    setSearching(false);
+  }, [displayedFiles]);
+
+  useEffect(() => {
+    setHasClipboardApi(Boolean(navigator?.clipboard));
+  }, [setHasClipboardApi]);
+
+  useEffect(() => {
+    if (!currentClip) return;
+    const timeout = window.setTimeout(() => {
+      setCurrentClip("");
+    }, 5000);
+    return () => {
+      if (timeout) window.clearTimeout(timeout);
+    };
+  }, [currentClip, setCurrentClip]);
+
+  function copyToClipboard(data) {
+    navigator.clipboard.writeText(data);
```

Comment:
`navigator.clipboard.writeText(data)` return a promise

Could we do 
```
navigator.clipboard.writeText(data).then(() => {
 // code to handle currentClip
}).finally();


```

Reply:
the code above is not about async issues it's all about removing the copied state after a while but it's not really necessary at all